### PR TITLE
Add stroke count option

### LINKED ISSUE #94: Add information to distinguish kanji variants
As I explained in the recent issue (#90), there's currently no way to distinguish between variant forms.

I suggest adding an option that shows the number of strokes so that in most cases I can know what form the answer is.

I think this is a good idea because things about itaiji are complicated and sometimes controversial.

### PR DESCRIPTION
**Pull request by ghost**
closes #94. I didn't include this in kana review, because I thought it wasn't necessary (i.e. あ and ぁ have the same number of strokes).

btw this is my first time I touched Java/Kotlin and IntelliJ 😗

<table>
<td>

scr_studypreview

</td>
<td>

scr_preview

</td>
</table>

### CODE REVIEW

File: core/src/commonMain/kotlin/ua/syt0r/kanji/presentation/common/resources/string/EnglishStrings.kt [hunk 1]
```diff
@@ -458,6 +458,13 @@ object EnglishWritingPracticeStrings : WritingPracticeStrings {
     override val leftHandedModeMessage: String =
         "Adjusts position of input in landscape mode of writing practice screen"
 
+    override val strokesMessage: (count: Int) -> String = {
+        buildString {
+            append(it.toString())
+            if (it == 1) append(" stroke")
+            else append(" strokes")
```
Comment (martinetd) [inline core/src/commonMain/kotlin/ua/syt0r/kanji/presentation/common/resources/string/EnglishStrings.kt]: (Far from a kotlin expert, but the standard way of doing this would be the `plurals` resource like this:
```
    <plurals name="writing_practice_summary_mistakes">
        <item quantity="one">%d mistake</item>
        <item quantity="other">%d mistakes</item>
    </plurals>
```

With that said we already have something like this function precisely for mistakes in another place:
```
    override val savingMistakesMessage: (count: Int) -> String = {
        if (it == 1) "1 mistake" else "$it mistakes"
    }
```
So it probably doesn't really matter either way)

File: app/src/main/java/ua/syt0r/kanji/presentation/preview/screen/WritingPracticeScreenPreview.kt [hunk 2]
```diff
@@ -25,7 +25,7 @@ import kotlin.time.Duration.Companion.seconds
 @Composable
 private fun KanjiPreview(
     darkTheme: Boolean = false,
-    isStudyMode: Boolean = true
+    isStudyMode: Boolean = false
```
Comment (martinetd) [inline app/src/main/java/ua/syt0r/kanji/presentation/preview/screen/WritingPracticeScreenPreview.kt:28]: (didn't check if this is ever used without parameter specified, but default value probably shouldn't change here? or if it does that looks like it'd belong to another commit)

**Review by martinetd — COMMENTED** [review]
Just a drive by comment as I was looking at open PR to get a first feel at the code -- feel free to ignore :)

File: app/src/main/java/ua/syt0r/kanji/presentation/preview/screen/WritingPracticeScreenPreview.kt (hunk 2 above)
Comment (AttractLight) [reply app/src/main/java/ua/syt0r/kanji/presentation/preview/screen/WritingPracticeScreenPreview.kt:28]: Maybe this was originally `false` and changed for debugging, or just mistake. Because there's a `KanjiStudyPreview` function which explicitly passes `isStudyMode` as true.

File: core/src/commonMain/kotlin/ua/syt0r/kanji/presentation/common/resources/string/EnglishStrings.kt (hunk 1 above)
Comment (syt0r) [reply core/src/commonMain/kotlin/ua/syt0r/kanji/presentation/common/resources/string/EnglishStrings.kt]: Yeah, I'm not using resources because it's a multiplatform project and I didn't want to introduce some 3rd party library since it can be done this way anyway

**Comment by syt0r** [comment]
I'm not completely sure about one thing, do you feel that stroke count might be a huge hint? I was thinking maybe it's worth adding only for kanji that have variants and make it not optional for this case

**Comment by AttractLight** [comment]
Personally I think stroke count wouldn't be a hint for composition. But anyway that's a better design. This should be closed if you prefer it. However some variant forms have the same number of strokes (e.g. 劔 and 劒), so in that case something other than stroke count should be shown. I'll be using this build as a workaround myself.

File: core/src/commonMain/kotlin/ua/syt0r/kanji/presentation/screen/main/screen/writing_practice/ui/WritingPracticeInfoSection.kt [hunk 3]
```diff
@@ -276,6 +299,14 @@ private fun AnimatedCharacterSection(
 
 }
 
+@Composable
+private fun StrokeCountText(count: Int) {
+    Text(
+        text = resolveString { writingPractice.strokesMessage(count) },
+        style = MaterialTheme.typography.bodyMedium.plus(TextStyle(color = Color.Gray))
```
Comment (syt0r) [inline core/src/commonMain/kotlin/ua/syt0r/kanji/presentation/screen/main/screen/writing_practice/ui/WritingPracticeInfoSection.kt]: Please set text color using color argument, not by modifying style. Also use color from theme like `MaterialTheme.colorScheme.onSurface` or something, you can use `copy(alpha =)` method of Color to make it less visible if available colors are not suitable

**Review by syt0r — COMMENTED** [review]
Good job with figuring out everything overall 😅
